packages/platforms/accton/x86-64/as5915-18x/onlp/builds/x86_64_accton_as5915_18x/module/src/sysi.c

```diff
@@ -150,3 +150,217 @@ int onlp_sysi_platform_manage_leds(void)
 
     return onlp_ledi_mode_set(ONLP_LED_ID_CREATE(LED_FAN), ONLP_LED_MODE_GREEN);
 }
+
+enum fan_duty {
+    FAN_DUTY_MAX = 100,    /* high speed */
+    FAN_DUTY_MID = 60,     /* middle speed */
+    FAN_DUTY_NOR = 30,     /* normal speed */
+    FAN_DUTY_LOW_1_3 = 0,  /* low speed for fan1 and fan3*/
+    FAN_DUTY_LOW_2_4 = 40  /* low speed for fan2 and fan4 */
+};
+
+enum thermal_threshold {
+    Tc = 78000,
+    Tm = 57000,
+    Tn = 50000,
+    Tb = 39000,
+    Ta = 32000,
+    T0 = 6000,
+    Te = 0,
+};
+
+static int
+sysi_fanctrl_fan_set_duty(int p)
+{
+    int i;
+    int status = 0;
+
+    for (i = 1; i <= CHASSIS_FAN_COUNT; i++) {
+        int ret = 0;
+
+        ret = onlp_fani_percentage_set(ONLP_FAN_ID_CREATE(i), p);
+        if (ret < 0) {
+            status = ret;
+        }
+    }
+
+    return status;
+}
+
+static int
+sysi_fanctrl_fan_set_duty_low(void)
+{
+    int i;
+    int status = 0;
+    int duty[CHASSIS_FAN_COUNT] = {FAN_DUTY_LOW_1_3, FAN_DUTY_LOW_2_4, FAN_DUTY_LOW_1_3, FAN_DUTY_LOW_2_4};
+
+    for (i = 1; i <= CHASSIS_FAN_COUNT; i++) {
+        int ret = 0;
+
+        ret = onlp_fani_percentage_set(ONLP_FAN_ID_CREATE(i), duty[i-1]);
+        if (ret < 0) {
+            status = ret;
+        }
+    }
+
+    return status;
+}
```

Comment:
The two functions could be refactored as they mostly do the same, and the array is passed as an argument. `sysi_fanctrl_fan_set_duty_low()` would then just call that function:

    sysi_fanctrl_fan_set_duty(FAN_DUTY_LOW_1_3, FAN_DUTY_LOW_2_4, FAN_DUTY_LOW_1_3, FAN_DUTY_LOW_2_4)

Reply:
Modified.